megamek/src/megamek/client/bot/princess/BasicPathRanker.java

```diff
@@ -958,8 +958,14 @@ private double calcWaterHazard(Entity movingUnit, Hex hex, MoveStep step,
             return UNIT_DESTRUCTION_FACTOR;
         }
 
+        MoveStep lastStep = movePath.getLastStep();
         // Unsealed unit will drown.
-        if (movingUnit instanceof Mech && ((Mech) movingUnit).isIndustrial()) {
+        if (movingUnit instanceof Mech
+                && ((Mech) movingUnit).isIndustrial()
+                && !movingUnit.hasEnvironmentalSealing()
+                && (movingUnit.getEngine().getEngineType() == Engine.COMBUSTION_ENGINE)
+                && hex.depth() >= 2
+                && step.equals(lastStep)) {
```

Comment:
So a mech can move through super deep water and not flood the engine as long as it ends the movement with the engine above water?

Reply:
The full rules from the footnote on TW page 52 is interesting, and unusual compared to the rules for various other non-fully-sealed units winding up underwater. To stay underwater indefinitely, Industrialmechs need environmental sealing and an underwater-capable engine type. However, IMs that _don't_ meet these requirements can still enter a hex where they will be fully underwater, but they are destroyed if they _remain_ fully underwater in the end phase of the turn _after_ they initially entered the water. In other words, a non-sealed ICE IM could walk into depth 2 water and end its movement there on one turn and survive as long as it managed to climb out the next turn. They would be considered destroyed if they failed to leave the deep water after one turn.

Reply:
Heh, so maybe then the criteria should be "if the move path both starts and ends under water", rather than just "if the move path ends under water".

Reply:
I think going into depth 2 water is also dangerous, so it should be weighted also.    you many not be allowed to move out of it and end up dead earlier than if you were out of water.

I should also add a smaller weight for going into depth 1 water, since if you then go prone there for 2 rounds you will die.